File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 5/24/2025, 1:50:39 AM
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_price=float('inf')
        max_profit=0
        for price in prices:
          if price<min_price:
            min_price = price
          potential_profit = price - min_price

          if potential_profit>max_profit:
             max_profit=potential_profit
        return max_profit

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.java
// Last updated: 5/24/2025, 1:50:32 AM
class Solution {
    public int maxProfit(int[] prices) {
        int lsf=Integer.MAX_VALUE;
        int op = 0;
        int pist = 0;

        for(int i =0;i<prices.length; i++){
            if(prices[i]<lsf){
                lsf=prices[i];
            }
            pist = prices[i]-lsf;
            if(op<pist){
                op = pist;
            }
        }
        return op;
    }
}
